ddl/ddl_api.go

```diff
@@ -1925,6 +1925,9 @@ func (d *ddl) CreateTableWithInfo(
 // preSplitAndScatter performs pre-split and scatter of the table's regions.
 // If `pi` is not nil, will only split region for `pi`, this is used when add partition.
 func (d *ddl) preSplitAndScatter(ctx sessionctx.Context, tbInfo *model.TableInfo, pi *model.PartitionInfo) {
+	if tbInfo.TempTableType != model.TempTableNone {
+		return
```

Comment:
It's better to report an error.
And thus, you can test it by checking an error in `create table t(id int ) pre_split_regions=2;`.